cmd/commands/run.go

```diff
@@ -147,20 +255,113 @@ func generateEvent(ctx context.Context, fn function.Function, eventName string)
 
 		data := event.Event{}
 		err := json.Unmarshal(evt, &data)
-		return data, err
+		return []event.Event{data}, err
+	}
+
+	fakedEvent, err := fakeEvent(ctx, fn, triggerName)
+	if err != nil {
+		return nil, err
 	}
 
-	return fakeEvent(ctx, fn, eventName)
+	return []event.Event{fakedEvent}, nil
 }
 
 // fakeEvent finds event triggers within the function definition, then chooses
 // a random trigger from the definitions and generates fake data for the event.
-func fakeEvent(ctx context.Context, fn function.Function, eventName string) (event.Event, error) {
+func fakeEvent(ctx context.Context, fn function.Function, triggerName string) (event.Event, error) {
 	triggers := []function.Trigger{}
 	for _, t := range fn.Triggers {
-		if t.EventTrigger != nil && (eventName == "" || eventName == t.EventTrigger.Event) {
+		if t.EventTrigger != nil && (triggerName == "" || triggerName == t.EventTrigger.Event) {
 			triggers = append(triggers, t)
 		}
 	}
 	return function.GenerateTriggerData(ctx, runSeed, triggers)
 }
+
+// Fetch `count` latest `triggerName` events from the event store.
+//
+// Requires triggerName is not empty, and count is >0.
+func fetchRecentEvents(ctx context.Context, triggerName string, count int64) ([]event.Event, error) {
+	if triggerName == "" {
+		return nil, fmt.Errorf("triggerName is required")
+	}
+
+	if count <= 0 {
+		return nil, fmt.Errorf("count must be >0")
+	}
+
+	s := state.RequireState(ctx)
+
+	ws, err := state.Workspace(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	archivedEvents, err := s.Client.RecentEvents(ctx, ws.ID, triggerName, count)
+	if err != nil {
+		return nil, err
+	}
+
+	events := []event.Event{}
+
+	for _, archivedEvent := range archivedEvents {
+		type evtData struct {
+			Data map[string]interface{}
```

Comment:
Is this missing the `User` `map[string]interface{}` and `v` as well to be parsed or were those intentionally left off? ([related docs](https://www.inngest.com/docs/event-format-and-structure)). This would be the same comment for `fetchEvent` below.

Reply:
Yep good spot! Added.